### PR TITLE
build: Add Lint Workflows for `cli` and `sdk-node`

### LINKED ISSUE #39: Create lint workflow for CLI
## Description
The feature should follow exactly the same flow as `web.yaml`. Please take refs from that. Depends on #22 

To summarize,
- [ ] Create a `cli.yaml`
- [ ] Make sure it runs only on pull requests and pushes
- [ ] Also, make sure the entire workflow executes only when changes are made to `apps/cli`

### PR DESCRIPTION
**Pull request by HarshPatel5940**
## Type
enhancement

___

## Description
This PR introduces two new GitHub Actions workflows for the CLI and SDK-Node packages. The workflows are triggered on push and pull request events that affect the respective package paths. Each workflow includes steps for:
- Checking out the code
- Setting up Node.js
- Installing pnpm
- Installing packages
- Running lint
- Running tests

___

## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>cli.yaml&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        .github/workflows/cli.yaml<br><br>

**This file has been added to define a new GitHub Actions <br>workflow for the CLI package. The workflow is triggered on <br>push and pull request events that affect the 'apps/cli/**' <br>path. It includes steps for checking out the code, setting <br>up Node.js, installing pnpm, installing packages, running <br>lint, and running tests.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/46/files#diff-b186cdccd59addc2f6171a64b1c1e97f5759efec23139108f6702ce661b1d370"> +39/-0</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>sdk-node.yaml&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        .github/workflows/sdk-node.yaml<br><br>

**This file has been added to define a new GitHub Actions <br>workflow for the SDK-Node package. The workflow is triggered <br>on push and pull request events that affect the <br>'packages/sdk-node' path. It includes steps for checking out <br>the code, setting up Node.js, installing pnpm, installing <br>packages, running lint, and running tests.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/46/files#diff-d99201f37b380fbef88d164d988ac3e55e384ae6c4ef1255a93e568d4633e76c"> +37/-0</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

## Description
This pull request adds a lint workflow for the CLI and SDK-Node packages. 

## Possible Improvements
Can we add a pnpm cache store for faster install time
```
- name: Get pnpm store directory
  shell: bash
  run: |
    echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV

- uses: actions/cache@v3
  name: Setup pnpm cache
  with:
    path: ${{ env.STORE_PATH }}
    key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
    restore-keys: |
      ${{ runner.os }}-pnpm-store-
      
```

Fixes #39 #40 
Parent Issue #36 

## Mentions
@rajdip-b 

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work
- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings

- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [ ] There are no UI/UX issues

### CODE REVIEW

**Comment by HarshPatel5940** [comment]
Let me know if we can use the pnpm cache store as i demonstrated in the PR description

**Comment by rajdip-b** [comment]
Hmmm I'm totally unaware about the pnpm cache. I guess we can put it to use once, and see if its ok

**Comment by HarshPatel5940** [comment]
> Hmmm I'm totally unaware about the pnpm cache. I guess we can put it to use once, and see if its ok
> 

Ok. Noted.

**Comment by rajdip-b** [comment]
@HarshPatel5940 hey, i think that the previous commit you made, had some formatting errors. please make sure that your workflow files are indented properly.
image

**Comment by HarshPatel5940** [comment]
> @HarshPatel5940 hey, i think that the previous commit you made, had some formatting errors. please make sure that your workflow files are indented properly.
> image
> 

Oh shit, i will be careful from next time.

Fixing it.

File: .github/workflows/api.yaml [hunk 1]
```diff
@@ -23,6 +23,19 @@ jobs:
         with:
           version: 8
           run_install: false
+          
+      - name: Get pnpm store directory
```
Comment (rajdip-b) [inline .github/workflows/api.yaml:27]: I think this one in here is out of alignment

File: .github/workflows/api.yaml [hunk 2]
```diff
@@ -23,6 +23,19 @@ jobs:
...
+        shell: bash
+        run: |
+          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
+
+      - uses: actions/cache@v3
```
Comment (rajdip-b) [inline .github/workflows/api.yaml:32]: This too

**Review by rajdip-b — CHANGES_REQUESTED** [review]
Please make the changes :D

File: .github/workflows/api.yaml (hunk 1 above)
Comment (HarshPatel5940) [reply .github/workflows/api.yaml:27]: got it!

File: .github/workflows/api.yaml (hunk 2 above)
Comment (HarshPatel5940) [reply .github/workflows/api.yaml:32]: got it.

**Comment by HarshPatel5940** [comment]
resolved comments.

**Review by rajdip-b — APPROVED** [review]

**Comment by HarshPatel5940** [comment]
I am also guessing we should do this, as i can't other actions running. what do you think?

```diff
-    paths: ['apps/web/**', '.github/workflows/web.yaml']
+    paths: ['apps/web/**', './github/workflows/web.yaml']
```

**Comment by rajdip-b** [comment]
> I am also guessing we should do this, as i can't see other actions running. what do you think?
> 
> ```diff
> -    paths: ['apps/web/**', '.github/workflows/web.yaml']
> +    paths: ['apps/web/**', './github/workflows/web.yaml']
> ```

Which ones btw? Because last I checked, they were running!

**Comment by HarshPatel5940** [comment]
> > I am also guessing we should do this, as i can't see other actions running. what do you think?
> > ```diff
> > -    paths: ['apps/web/**', '.github/workflows/web.yaml']
> > +    paths: ['apps/web/**', './github/workflows/web.yaml']
> > ```
> 
> Which ones btw? Because last I checked, they were running!

I am talking about actions not running in this PR even after we update the workflow file

**Comment by rajdip-b** [comment]
If you are talking about the cli and sdk workflows, then yes, they wont run right now. because they still are not a part of the codebase

**Comment by rajdip-b** [comment]
and for the PR, im fixing it rn.

**Comment by HarshPatel5940** [comment]
Ok. Thank you for helping me!

**Review by rajdip-b — APPROVED** [review]

**Pull request merged**